activerecord/lib/arel/nodes/fragments.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Arel # :nodoc: all
+  module Nodes
+    class Fragments < Arel::Nodes::Node
+      attr_reader :values
+
+      def initialize(values)
+        super()
+        @values = values
+      end
+
+      def initialize_copy(other)
+        super
+        @values = @values.clone
+      end
+
+      def hash
+        [@values].hash
+      end
+
```

Comment:
I think it might be useful to have one of these:

```
def <<(node)
  @values << node
end
```

... and perhaps even give `values` a default of `[]` in the constructor?

Reply:
This is a wonderful idea. Done!